vcs.mk

```diff
@@ -11,34 +11,56 @@ endif
 CLOCK_PERIOD ?= 1.0
 RESET_DELAY ?= 777.7
 
+#----------------------------------------------------------------------------------------
+# gcc configuration/optimization
+#----------------------------------------------------------------------------------------
+# -flto slows down compilation on small-memory and breaks on firesim-manager
+CMODE := -O3 -fbranch-probabilities -march=native
```

Comment:
Where did this come from? Is it known to always be better than the default? Is it optimizing for compilation time or runtime? Is that what we want?

Reply:
This was copying what Verilator did (though to make Verilator not segfault I just kept the `-O3`). It is optimizing the runtime. The default is nothing.

Reply:
Does it actually improve runtime?

Reply:
I dunno. I can run the bmark-tests on Rocket with and without the `CMODE` flags. Im trying to get Hammer sim tested 1st.

Reply:
With CMODE flags:
Compile: 36s
Runtime for all bmark tests (serially): 7m 36s

Without CMODE flags:
Compile: 35s
Runtime for all bmark tests (serially): ???

Reply:
While this doesn't result in a large runtime difference, I think it should stay if we end up consolidating flags across VCS and Verilator.

Reply:
Fine. It is worth noting that adding code that doesn't affect things is confusing, but if this proves that we can merge the flags in the future I'm fine with leaving it in.